File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float change;
    do
    {
        change = get_float("Change owed:");                     //get input "change", for changed owed
    }
    while (change <= 0);                                        //malicious user check

    int cents = round (change * 100);                           //round change*100 (to convert float "change" to interger "cents")

    int q = 0;                                                  //set quarter count to zero
    int d = 0;                                                  //set dime count to zero
    int n = 0;                                                  //set nickle count to zero
    int p = 0;                                                  //set penny count to zero

        do
        if (cents >= 25)                                        //quarter counter
        {
            cents = cents - 25;
            q++;
        }
        while (cents >= 25 && cents > 0);

        do
        if (cents >= 10)                                        //dime counter
        {
            cents = cents - 10;
            d++;
        }
        while (cents >= 10 && cents > 0);

        do
        if (cents >= 5)                                         //nickle counter
        {
            cents = cents - 5;
            n++;
        }
        while (cents >= 5 && cents > 0);

        do
        if (cents >= 1)                                         //penny counter
        {
            cents = cents - 1;
            p++;
        }
        while (cents >= 1 && cents > 0);

    int coins = q + d + n + p;                                  //set "coins" to total amount of coin used
    printf("%i\n", coins);                                      //print "coin" for number of coins used output
}

File: pset6/cash/cash.py
from cs50 import get_float
change = 0
while change <= 0:
    change = get_float("Change owed: ")
cents = change * 100
q = 0
d = 0
n = 0
p = 0
while cents >= 25:
    cents -= 25
    q += 1
while cents >= 10:
    cents -= 10
    d += 1
while cents >= 5:
    cents -= 5
    n += 1
while cents >= 1:
    cents -= 1
    p += 1
coins = q + d + n + p
print(f"{coins}")